handlers/public.go

```diff
@@ -167,3 +170,11 @@ func getPaymentResource(w http.ResponseWriter, req *http.Request, resource strin
 	}
 	return paymentResource, nil
 }
+
+func generateRandomID() (i string) {
+	ranNumber := strconv.Itoa(1000000 + rand.Intn(9000000))
+	now := time.Now()
+	nanos := now.UnixNano()
+	millis := strconv.FormatInt((nanos / 1000000), 10)
```

Comment:
Can combine these lines to avoid assigning to vars (and also use `time.Millesecond` rather than a hard-coded value:
```
    now := strconv.FormatInt(time.Now().UnixNano()/int64(time.Millisecond), 10)
    return ranNumber + now
```